src/compiler/parser.ts

```diff
@@ -4411,7 +4411,16 @@ namespace ts {
 
         function parseAwaitExpression() {
             const pos = getNodePos();
-            return finishNode(factory.createAwaitExpression(nextTokenAnd(parseSimpleUnaryExpression)), pos);
+            nextToken(); // advance past "await"
+            const op = tryParse(() => {
+                if (token() !== SyntaxKind.DotToken) return undefined;
+                nextToken(); // advance past the dot
+                if (!tokenIsIdentifierOrKeyword(token())) return undefined;
+                const id = parseIdentifierName().escapedText;

```

Comment:
According to current spec, we should emit error when the identifier contains escape.
```js
await . r\u{61}ce
```
is not an await operation expression.

Reply:
it's an early prototype and I'm not familiar with TS tokenizer/scanner so won't fix in the current stage